File: leetcode/49. Group Anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
  let map = {};
  const answer = [];

  strs.forEach((s) => {
    let sorted_s = s.split("").sort().join("");
    if (!map[sorted_s]) {
      map[sorted_s] = [s];
    } else {
      map[sorted_s].push(s);
    }
  });
  Object.keys(map).forEach((key) => {
    answer.push(map[key]);
  });
  return answer;
};

File: leetcode/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        answer = []
        anaMap = {}

        for s in strs:
            cur = sorted(s)
            cur = "".join(cur)
            
            if cur not in anaMap:
                anaMap[cur] = [s]
            else:
                anaMap[cur].append(s)
        
        for key in anaMap:
            answer.append(anaMap[key])
        
        return answer